### ISSUE TITLE
Smoother terminology: commit, pull request, patch, changeset

### ISSUE THREAD

**Issue by nedbat** [python/devguide#1443]
### Describe the bug

These four terms are used in the devguide.  We should make the terminology uniform, and based on git/GitHub usage.

- I think we should do away with "changeset" completely.  
- "Patch" can often be changed to "commit" or "pull request" or even just "change."  We have a habit of saying "patch" in the change log: "Patch by Josephine Koder," I'm not sure what to do about that.  A few places use the term in the git sense, a diff applied textually, which I think is OK.

**Comment by ezio-melotti** [comment]
"changset" was the official term used in the Mercurial days, and it should be fine to get rid of those since Git uses the term "commit" instead.
In some contexts, "patch" was used to refer to the actual `.diff` files that we used to attach to the b.p.o issues before we moved to GitHub and started using PRs, and in those instances it can be replaced with the terms you suggested.  The change log initially used the term to refer to the `.diff` files, but since "patch" is also a more generic concept[^1], it still works. 

[^1]: e.g. in a sentence like "The vulnerability was patched by merging Josephine Koder's PR."
